### PR TITLE
Triple Oscillator GUI redesign

### LINKED ISSUE #2243: Triple OSC new UI proposal
Hello, I decided to redesign my old UI for Triple OSC. With some help of my brother here is it. I'm waiting for your advices and opinions. If you like it I can bring source files to implement it into LMMS.
triple
**EDIT** 
 > changed name from Triple OSC to full name (Triple Oscillator)
 > changed fonts
 > PD/SPD - fixed to PO/SPO
 > sine shape block rotated by 180 degree
 > some text aligns
triple
>added citrus
>changed sine and noise wave
>new user shape icon
synnth

### PR DESCRIPTION
**Pull request by LocoMatt**
Resolves #2243 

3oscgui

### CODE REVIEW

**Comment by tresf** [comment]
@LocoMatt this looks great.  I'm building it now.

P.S. Please issue a PR against https://github.com/lmms/artwork with your source files so that we have them for future theme changes.

**Comment by tresf** [comment]
So IMO, the small icon needs a stroke or something to give contrast against the brighter background.

image

Also, the top track icon is quite bright now when compared to the rest of the interface.

image


-Tres

**Comment by tresf** [comment]
Also, there's something wrong with the knob artwork on my build environment (Ubuntu 12.04 x64).  I'll do a make install to see if that helps.

image

File: data/themes/default/style.css [hunk 1]
```diff
@@ -551,9 +551,9 @@ BBTCOView {
 
 /* Plugins */
 
-TripleOscillatorView Knob {
-	color: rgb(1, 32, 64);
-	qproperty-outerColor: rgb(0, 0, 0);
+TripleOscillatorView knob {
```
Comment (grejppi) [inline data/themes/default/style.css]: @tresf this may be why it's breaking

Comment (Umcaruje) [inline data/themes/default/style.css]: Yup, it seems like `knob` should be `Knob` and that's what broke it.

@LocoMatt can you please fix this?

**Comment by tresf** [comment]
Two remaining items I haven't seen an answer on... 

 1. The top track icon is quite bright now when compared to the rest of the interface and could probably use a stroke or dropshadow to match the other icons as well.
 2. There's something wrong with the knob artwork on my build environment (Ubuntu 12.04 x64). I'll do a make install to see if that helps.

These issues have been open for 22 days now, please advise.

**Comment by LocoMatt** [comment]
@tresf Sorry for late but I was busy a bit last time. I've fixed knobs but I'm still not sure about the ogo. I noticed that last time some people started working on new theme and I'm not sure should I wait for them or not.

**Comment by tresf** [comment]
> not sure should I wait for them or not.

Integration with our existing theme is the most important.  I don't believe we'll be changing it out for the 1.2 release.

**Comment by tresf** [comment]
@LocoMatt have you had time to review the concerns in https://github.com/LMMS/lmms/pull/2293#issuecomment-134435518?  Once this is done, we can merge for the 1.2 release candidate!

**Comment by LocoMatt** [comment]
@tresf How about something like this?
shadow
shadow2

**Comment by Spekular** [comment]
I still think it's lacking contrast between slices, but maybe that's just
me. Have you tried making the slices darker the further back they are?

**Comment by LocoMatt** [comment]
shadow
shadow2
@Spekular @tresf

**Comment by Spekular** [comment]
Also the distance/rotation between the orange and lemon seems bigger than
the distance/rotation between the lemon and lime.

**Comment by Spekular** [comment]
That update definitely improved contrast imo! Thanks.

**Comment by LocoMatt** [comment]
Eventually if lemons doesn't fit i have this 
shadow
shadow2

**Comment by tresf** [comment]
> @tresf How about something like this?
shadow
shadow2

Much better.  Can you show the small version next to the others as well as the default template with it on the top next to the others?

**Comment by tresf** [comment]
Also, you may want to darken (or thicken) the lines between slices so they don't bleed into eachother as much at such a small size.

image

**Comment by tresf** [comment]
> Eventually if lemons doesn't fit i have this 
shadow
shadow2

Good to know we have a fallback, but the lemons seem to get crowd favorite, so I think we're in good tweaking what we have. :+1:

**Comment by Spekular** [comment]
@tresf he made a darker version already, btw.

**Comment by tresf** [comment]
> @tresf he made a darker version already, btw.

Can you elaborate?  My comments are based on the latest AFAIK.

**Comment by Spekular** [comment]
I'm gonna see if I can attach an image on mobile. You're responding to the
design that adds drop shadow, when there's a darkened/contrasty design
shortly after:

**Comment by tresf** [comment]
Thanks I see it now.  Tbe small rendering still blurs the lines.  I'll let @LocoMatt chime in.

**Comment by Spekular** [comment]
Ok that didn't work, here's an even worse workaround:
http://i.imgur.com/u2KZIHi.jpg

**Comment by Spekular** [comment]
Oh, you see it. Remind me to never browse github on mobile.

**Comment by LocoMatt** [comment]
@tresf @Spekular 
How about this?
shadow
shadow2

**Comment by tresf** [comment]
Terrific.  @LocoMatt did you ever fix the `knob` part discussed above?  I haven't tested this yet, but it could be merged otherwise.

**Comment by musikBear** [comment]
is it relevant in respect to the redesign to also remind about the decision to have a better and more musical setup on the wave-shape, and most important the ENV settings. It is such a small thing, but it is a recurring question from new-comers almost weekly : 
*why do i hear clicking noises *
One almost feel obliged to answer :
*because the setup of our default preset sucks*
I dont.

I cant find the topic specified for this anymore, so sorry, no reference -guess the tittle was changed

**Comment by tresf** [comment]
@musikBear,

Better or more sane ENV defaults are valid concern, but should be written up in a proper dedicated bug report.

**Comment by michaelgregorius** [comment]
Tested against the current master where it looks as follows:
triple-osc-redesign

Can this one be merged?

**Comment by tresf** [comment]
> Can this one be merged?

Your screenshots don't show the same icon per https://github.com/LMMS/lmms/pull/2293#issuecomment-141555766.

-Tres

**Comment by tresf** [comment]
Also, we should squash the commits into a single commit prior to merging.

**Comment by michaelgregorius** [comment]
@LocoMatt Can you update the icon and squash your commits so that we can merge this one? Thanks!

**Comment by Umcaruje** [comment]
Ok, so this was rebased and squashed. I built the branch and it looks like this: 
screenshot from 2016-01-27 00 13 38
screenshot from 2016-01-27 00 15 15

Also, Travis passed. 

Merging.

**Pull request merged**